src/main/java/com/mraof/minestuck/block/plant/GlowingMushroomVinesBlock.java

```diff
@@ -0,0 +1,47 @@
+package com.mraof.minestuck.block.plant;
+
+import com.mraof.minestuck.util.MSTags;
+import net.minecraft.core.BlockPos;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.Vec3;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class GlowingMushroomVinesBlock extends Block
+{
+	protected static final VoxelShape SHAPE = Block.box(3.0D, 0.0D, 3.0D, 13.0D, 16.0D, 13.0D);
+	
+	public GlowingMushroomVinesBlock(Properties properties)
+	{
+		super(properties);
+	}
+	
+	@Override
+	public void randomTick(BlockState state, ServerLevel level, BlockPos pos, RandomSource rand)
+	{
+		BlockPos belowPos = pos.below();
+		if(rand.nextInt(90) == 0 && level.getBlockState(belowPos).isAir())
+		{
+			level.setBlock(belowPos, this.defaultBlockState(), Block.UPDATE_ALL);
+		}
+	}
+	
+	@Override
+	public boolean canSurvive(BlockState state, LevelReader level, BlockPos pos)
+	{
+		BlockState aboveState = level.getBlockState(pos.above());
+		return aboveState.is(this) || aboveState.is(MSTags.Blocks.SHADEWOOD_LEAVES);
+	}
```

Comment:
The implementation of this has issues. if the block above is broken the vine does not drop for whatever reason